tools/src/main/java/org/apache/kafka/trogdor/workload/SustainedConnectionWorker.java

```diff
@@ -0,0 +1,610 @@
+    }
+
+    private interface SustainedConnection extends AutoCloseable {
+        boolean needsRefresh();
+        void refresh();
+        void claim();
+    }
+
+    private class MetadataSustainedConnection implements SustainedConnection {
+
+        // This variable is used to maintain the connection itself.
+        private AdminClient client;
+
+        // These variables are used to maintain the scheduling of the connection keep-alive.
+        private long nextUpdate;
+        private boolean inUse;
+        private long refreshRate;
+
+        // This variable is used to maintain the connection properties.
+        private final Properties props;
+
+        MetadataSustainedConnection(SustainedConnectionSpec spec) {
+            // This variable is used to maintain the connection itself.
+            this.client = null;
+
+            // These variables are used to maintain the scheduling of the connection keep-alive.
+            this.nextUpdate = 0;
+            this.inUse = false;
+            this.refreshRate = spec.refreshRateMs();
+
+            // This variable is used to maintain the connection properties.
+            this.props = new Properties();
+            this.props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, spec.bootstrapServers());
+            WorkerUtils.addConfigsToProperties(this.props, spec.commonClientConf(), spec.commonClientConf());
+        }
+
+        @Override
+        public void refresh() {
+            try {
+                if (this.client == null) {
+
+                    // Housekeeping to track the number of opened connections.
+                    synchronized (SustainedConnectionWorker.this) {
+                        SustainedConnectionWorker.this.totalMetadataConnections += 1;
+                    }
+
+                    // Create the admin client connection.
+                    this.client = AdminClient.create(this.props);
+                }
+
+                // Fetch some metadata to keep the connection alive.
+                this.client.describeCluster().nodes().get();
+
+            } catch (Throwable e) {
+
+                // Housekeeping to track the number of opened connections and failed connection attempts.
+                synchronized (SustainedConnectionWorker.this) {
+                    SustainedConnectionWorker.this.totalMetadataConnections -= 1;
+                    SustainedConnectionWorker.this.totalMetadataFailedConnections += 1;
+                }
+
+                SustainedConnectionWorker.log.error(e.getMessage());
+
+                // Set the admin client to be recreated on the next cycle.
+                this.closeQuietly();
+            }
+
+            // Schedule this again and set to not in use.
+            this.nextUpdate = System.currentTimeMillis() + this.refreshRate;
+            this.inUse = false;
+        }
+
+        @Override
+        public boolean needsRefresh() {
+            return !this.inUse && (System.currentTimeMillis() > this.nextUpdate);
```

Comment:
I see a fair amount of code duplication with all the interface method calls - maybe we can create an abstract class to hold these reused methods?

Reply:
Done.